packages/x-components/src/components/panels/base-tabs-panel.vue

```diff
@@ -0,0 +1,331 @@
+<template>
+  <section v-if="getTabs().length > 0" class="x-tabs-panel" data-test="base-tabs-panel">
+    <component
+      :is="animation"
+      tag="ul"
+      class="x-list"
+      :class="tabsListClass"
+      data-test="base-tabs-panel-list"
+    >
+      <li v-for="tab in getTabs()" :key="tab">
+        <!--
+          @slot Slot that is be used for replacing the whole tab.
+            @binding {tab} string - The tab name.
+            @binding {isSelected} boolean - Indicates if the tab is selected.
+            @binding {selectTab} {() => void} selectTab - Function to select the tab.
+         -->
+        <slot
+          name="tab"
+          v-bind="{ tab, isSelected: tabIsSelected(tab), selectTab: () => selectTab(tab) }"
+        >
+          <button
+            :key="tab"
+            @click="selectTab(tab)"
+            class="x-button x-tabs-panel__button"
+            :class="tabIsSelected(tab) ? activeTabClass : tabClass"
```

Comment:
Should we apply the `tabClass` always, and the `activeTabClass` only when selected?
```suggestion
            :class="[tabClass, { [activeTabClass]: tabIsSelected(tab) }]"
```

Reply:
It's a good question, my first approach was to always add `tabClass` but I think it can be a problem under some scenarios. 

For example, having the following values:
```
<BaseTabsPanel
  activeTabClass="x-button-auxiliary x-button-lg" <!-- Larger button with auxiliary style -->
  tabClass="x-button-ghost x-button-sm" <!-- Smaller button with ghost style -->
>
```
This would lead to the active tab having the following classes: `x-button-ghost x-button-sm x-button-auxiliary x-button-lg`, where some styles for the active tab can overlap with the tab styles, to overcome this we would need to maybe use the `!important` flag.

Reply:
I think that it is better to apply only one of them to avoid having the situations that @mnavarroespinosa describes.